zebra/zebra_dplane.c

```diff
@@ -258,6 +259,43 @@ struct dplane_rule_info {
 	struct dplane_ctx_rule old;
 };
 
+struct dplane_netfilter_info {
```

Comment:
It might clean up the accessor apis if this just held a zebra_pbr_ipset_entry and maybe _table,  instead of a large number of individual attributes?

Reply:
Actually, I dont think; because when you handle ipset attributes, you may need to know on which iptable entry it is referenced. This is why I reused the same structure to handle both ipset and iptable objects.
any objections?

Reply:
what I meant was that you've got code that moves one attribute at a time to 'init' the fields here, and then accessors that access one field at a time to reconstruct ... an ipset, iptable, etc for processing. if you just did a union of the three structs here, populated  them directly in the 'init' step, and then had three accessors to re-populate them, you'd save a lot of lines? you might still need special handling for things like the array of names, but you'd reduce the footprint quite a bit.

Reply:
ok, I got what you mean.
let restart from scratch by using 3 new apis :
dplane_iptable_get()
dplane_ipset_get()
dplane_ipset_entry_get()